addons/membership/models/membership.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import api, models, fields, _
+import odoo.addons.decimal_precision as dp
+from odoo.exceptions import UserError
+
+STATE = [
+    ('none', 'Non Member'),
+    ('canceled', 'Cancelled Member'),
+    ('old', 'Old Member'),
+    ('waiting', 'Waiting Member'),
+    ('invoiced', 'Invoiced Member'),
+    ('free', 'Free Member'),
+    ('paid', 'Paid Member'),
+]
+
+
+class MembershipLine(models.Model):
+    _name = 'membership.membership_line'
+    _description = __doc__
+    _rec_name = 'partner_id'
+    _order = 'id desc'
+    partner_id = fields.Many2one('res.partner', 'Partner', ondelete='cascade', index=1, oldname='partner')
+    membership_id = fields.Many2one('product.product', string="Membership", required=True)
+    date_from = fields.Date('From', readonly=True)
+    date_to = fields.Date('To', readonly=True)
+    date_cancel = fields.Date('Cancel date')
+    date = fields.Date('Join Date', help="Date on which member has joined the membership")
```

Comment:
change field name to "date_join"  with keep oldname="date" and make separate commit too....